File: benchmarks/language-comparison/workload.py
def fib(n: int) -> int:
    if n <= 1:
        return n

    a = 0
    b = 1
    i = 2
    while i <= n:
        nxt = a + b
        a = b
        b = nxt
        i += 1
    return b


def fib_work(n: int, repeat_count: int) -> int:
    total = 0
    i = 0
    while i < repeat_count:
        total += fib(n)
        i += 1
    return total


def sum_squares(n: int) -> int:
    total = 0
    i = 1
    while i <= n:
        total += i * i
        i += 1
    return total


def list_push_sum(n: int) -> int:
    values = []
    i = 0
    while i < n:
        values.append(i * 3 + 7)
        i += 1

    total = 0
    j = 0
    while j < len(values):
        total += values[j]
        j += 1
    return total


def main() -> None:
    fib_acc = fib_work(32, 80_000)
    sum_acc = sum_squares(200_000)
    list_acc = list_push_sum(80_000)
    score = fib_acc + sum_acc + list_acc

    print(f"fib_acc={fib_acc}")
    print(f"sum_squares={sum_acc}")
    print(f"list_push_sum={list_acc}")
    print(f"score={score}")


if __name__ == "__main__":
    main()

File: benchmarks/language-comparison/workload.c
#include <stdint.h>
#include <stdio.h>
#include <stdlib.h>

static int64_t fib(int64_t n) {
    if (n <= 1) {
        return n;
    }

    int64_t a = 0;
    int64_t b = 1;
    int64_t i = 2;
    while (i <= n) {
        int64_t next = a + b;
        a = b;
        b = next;
        i += 1;
    }
    return b;
}

static int64_t fib_work(int64_t n, int64_t repeat_count) {
    int64_t total = 0;
    int64_t i = 0;
    while (i < repeat_count) {
        total += fib(n);
        i += 1;
    }
    return total;
}

static int64_t sum_squares(int64_t n) {
    int64_t total = 0;
    int64_t i = 1;
    while (i <= n) {
        total += i * i;
        i += 1;
    }
    return total;
}

static int64_t list_push_sum(int64_t n) {
    int64_t *values = (int64_t *)malloc((size_t)n * sizeof(int64_t));
    if (values == NULL) {
        return -1;
    }

    int64_t i = 0;
    while (i < n) {
        values[i] = i * 3 + 7;
        i += 1;
    }

    int64_t total = 0;
    int64_t j = 0;
    while (j < n) {
        total += values[j];
        j += 1;
    }

    free(values);
    return total;
}

int main(void) {
    int64_t fib_acc = fib_work(32, 80000);
    int64_t sum_acc = sum_squares(200000);
    int64_t list_acc = list_push_sum(80000);
    int64_t score = fib_acc + sum_acc + list_acc;

    printf("fib_acc=%lld\n", (long long)fib_acc);
    printf("sum_squares=%lld\n", (long long)sum_acc);
    printf("list_push_sum=%lld\n", (long long)list_acc);
    printf("score=%lld\n", (long long)score);
    return 0;
}
